jgrapht-core/src/test/java/org/jgrapht/graph/concurrent/AsSynchronizedGraphTest.java

```diff
@@ -338,10 +344,12 @@ public void execute()
             g.setCache(!g.isCacheEnabled());
```

Comment:
So sorry for the bug. This method should be synchronized because two visits of the graph may be interrupted.